### PR TITLE
[IPU] bert-base use paddle branch release/2.3

### PR DESCRIPTION
**Pull request by gglin001**
### PR types

Others
### PR changes

Models
### Description

1. use paddle branch `release/2.3`
2. add `compilation_progress_logger`, print compile progress

### CODE REVIEW

**Comment by ZeyuChen** [comment]
@gglin001 Hi Allen, we will release PaddleNLP v2.3 in 5.18. Glad to add IPU support as our new feature.
@guoshengCS Please add BERT IPU support into our v2.3 formal release note before 5.18

**Review by qili93 — DISMISSED** [review]
LGTM

File: examples/language_model/bert/static_ipu/custom_ops/custom_nll_loss.cc [hunk 1]
```diff
@@ -43,7 +43,7 @@ std::vector<paddle::Tensor> OpBackward(const paddle::Tensor &x) { return {x}; }
 PD_BUILD_OP(custom_nll_loss)
     .Inputs({"X", "Y"})
     .Outputs({"Out"})
-    .Attrs({"reduction: int", "ignoreIndex: int",
+    .Attrs({"reduction: int", "ignoreIndex: std::string",
```
Comment (gglin001) [inline examples/language_model/bert/static_ipu/custom_ops/custom_nll_loss.cc:46]: 这里的修改对应到 paddle release/2.3 的PR https://github.com/PaddlePaddle/Paddle/pull/42596

**Review by guoshengCS — APPROVED** [review]
LGTM. Thansks for the contribution! We have put it into the release note of PaddleNLP v2.3rc0, and also would make it on that of PaddleNLP v2.3.

**Pull request merged**